libs/sdk-ui-dashboard/src/model/store/catalog/catalogSelectors.ts

```diff
@@ -340,3 +388,42 @@ function getAttributesWithHierarchyDescendants(
     });
     return attributeDescendants;
 }
+
+function buildAdhocDateHierarchies(
+    dateHierarchyTemplates: IDateHierarchyTemplate[],
+    catalogDateDatasets: ICatalogDateDataset[],
+): ICatalogAttributeHierarchy[] {
+    const adhocDateHierarchies: ICatalogAttributeHierarchy[] = [];
+    catalogDateDatasets.forEach((dateDataset) => {
+        const dateAttributes = dateDataset.dateAttributes;
+        const defaultDateHierarchyTemplate = dateHierarchyTemplates.find(
+            (template) => template.id === "default",
+        );
+        const dateAttributesInHierarchy: ObjRef[] = [];
+        defaultDateHierarchyTemplate?.granularities.forEach((templateGranularity) => {
+            const dateAttribute = dateAttributes.find((attr) => attr.granularity === templateGranularity);
+            if (dateAttribute) {
+                dateAttributesInHierarchy.push(dateAttribute.attribute.ref);
+            }
+        });
+        if (!isEmpty(dateAttributesInHierarchy)) {
+            const adhocDateHierarchyId = ADHOC_DATE_HIERARCHY_PREFIX + dateDataset.dataSet.id;
+            adhocDateHierarchies.push({
+                type: "attributeHierarchy",
+                attributeHierarchy: {
+                    type: "attributeHierarchy",
```

Comment:
I'm not sure we should use the same type of object as for attribute hierarchy. The type would be better as "dateAttributeHierarchy" and `id`, `uri`, and `ref` are not necessary. Same goes for `title` and `description`. These texts are not localized and are hardcoded and my guess is that they will be displayed in tooltip bubble or elsewhere. I would keep `title` and set it to the name of date dataset.

That is the reason why specification mentions that these date hierarchies will be blacklisted based on date dataset attribute ID and date template ID, not ad hoc hierarchy ID.

Reply:
Do you mean that we will introduce a new type of object for date hierarchy instead of reusing the IAttributeHierarchyMetadataObject?

Reply:
fixed

Reply:
Hi @xMort, I have tried to introduce a new object type for date hierarchy which doesn't extend the metadata object as below
`export interface ICatalogDateAttributeHierarchy {
    type: "dateAttributeHierarchy";
    ref: ObjRef;
    title: string;
    attributes: ObjRef[];
}
`
I've noticed that we need to check whether the hierarchy is attribute or date in several places. If this side effect is  acceptable to you, I will update the PR with new approach.

Reply:
updated with new approach